homeassistant/components/datadog/__init__.py

```diff
@@ -43,63 +48,84 @@
 )
 
 
-def setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Set up the Datadog component."""
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the Datadog component from YAML config."""
+    if DOMAIN not in config:
+        return True
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": "import"},
```

Comment:
```suggestion
            context={"source": SOURCE_IMPORT},
```
`from homeassistant.config_entries import SOURCE_IMPORT`